ClimsoftVer4/ClimsoftVer4/frmNewFormDaily2.vb

```diff
@@ -0,0 +1,290 @@
+            ' temporary until we know how to get all fields from table without specifying names
+            dctSequencerFields.Add("elementId", New List(Of String)({"elementId"}))
+            ucrDaiy2Navigation.NewSequencerRecord(strSequencer:=txtSequencer.Text, dctFields:=dctSequencerFields, lstDateIncrementControls:=New List(Of ucrDataLinkCombobox)({ucrMonth}), ucrYear:=ucrYearSelector)
+        End If
+
+        ucrFormDaily.Focus()
+    End Sub
+
+    Private Sub btnCommit_Click(sender As Object, e As EventArgs) Handles btnCommit.Click
+        Try
+            If Not ValidateValues() Then
+                Exit Sub
+            End If
+
+            'Confirm if you want to continue and save data from key-entry form to database table
+            If MessageBox.Show("Do you want to continue and commit to database table?", "Save Record", MessageBoxButtons.YesNo, MessageBoxIcon.Question) = DialogResult.Yes Then
+                ucrFormDaily.SaveRecord()
+                ucrDaiy2Navigation.GoToNewRecord()
+                SaveEnable()
+                MessageBox.Show("New record added to database table!", "Save Record", MessageBoxButtons.OK, MessageBoxIcon.Information)
+            Else
+                MessageBox.Show("Record not Saved", "Save Record", MessageBoxButtons.OK, MessageBoxIcon.Information)
+            End If
+        Catch ex As Exception
+            MessageBox.Show("New Record has NOT been added to database table. Error: " & ex.Message, "Save Record", MessageBoxButtons.OK, MessageBoxIcon.Error)
+        End Try
+    End Sub
+
+    Private Sub btnUpdate_Click(sender As Object, e As EventArgs) Handles btnUpdate.Click
+        Try
+            If Not ValidateValues() Then
+                Exit Sub
+            End If
+
+            If Not chkRepeatEntry.Checked Then
+                If MessageBox.Show("Are you sure you want to update this record?", "Update Record", MessageBoxButtons.YesNo, MessageBoxIcon.Question) = DialogResult.Yes Then
+                    ucrFormDaily.SaveRecord()
+                    MessageBox.Show("Record updated successfully!", "Update Record", MessageBoxButtons.OK, MessageBoxIcon.Information)
+                End If
+            End If
+
+            ' When in double entry mode just skip update action. Values have been updated on entry
+            If chkRepeatEntry.Checked Then
+                Dim dctSequencerFields As New Dictionary(Of String, List(Of String))
+
+                'temporary until we know how to get all fields from table without specifying names
+                dctSequencerFields.Add("mm", New List(Of String)({"mm"}))
+                dctSequencerFields.Add("dd", New List(Of String)({"dd"}))
+                dctSequencerFields.Add("hh", New List(Of String)({"hh"}))
+                ucrDaiy2Navigation.NewSequencerRecord(strSequencer:=txtSequencer.Text, dctFields:=dctSequencerFields, lstDateIncrementControls:=New List(Of ucrDataLinkCombobox)({ucrMonth}), ucrYear:=ucrYearSelector)
```

Comment:
With the new implementation of connecting to the database, when `NewSequencerRecord` is called, a new record will be selected. Changes to the current record are updated in the entity framework but will not be sent to the database until `ucrFormDaily.SaveRecord()` is called so at this point in the code, it appears to me that changes to the current record will be lost when `chkRepeatEntry` is checked since `ucrFormDaily.SaveRecord()` is not called in this case.